application/src/main/java/org/togetherjava/tjbot/commands/help/HelpThreadCommand.java

```diff
@@ -0,0 +1,212 @@
+import java.util.function.Supplier;
+
+import static org.togetherjava.tjbot.commands.help.HelpSystemHelper.TITLE_COMPACT_LENGTH_MAX;
+import static org.togetherjava.tjbot.commands.help.HelpSystemHelper.TITLE_COMPACT_LENGTH_MIN;
+
+/**
+ * Implements the {@code /help-thread} command, which are contains special command for help threads
+ * only
+ */
+public final class HelpThreadCommand extends SlashCommandAdapter {
+    private static final int COOLDOWN_DURATION_VALUE = 30;
+    private static final ChronoUnit COOLDOWN_DURATION_UNIT = ChronoUnit.MINUTES;
+    private static final String CATEGORY = "category";
+    private static final String TITLE = "title";
+    private static final String CLOSE = "close";
+    private static final String BOOKMARK = "bookmark";
+
+    private static final Supplier<Cache<Long, Instant>> newCaffeine = () -> Caffeine.newBuilder()
+        .maximumSize(1_000)
+        .expireAfterAccess(COOLDOWN_DURATION_VALUE, TimeUnit.of(COOLDOWN_DURATION_UNIT))
+        .build();
+
+    private final HelpSystemHelper helper;
+    private final Cache<Long, Instant> helpThreadIdToLastCategoryChange = newCaffeine.get();
+    private final Cache<Long, Instant> helpThreadIdToLastTitleChange = newCaffeine.get();
+    private final Cache<Long, Instant> helpThreadIdToLastClose = newCaffeine.get();
+
+    /**
+     * Creates a new instance.
+     *
+     * @param config the config to use
+     * @param helper the helper to use
+     */
+    public HelpThreadCommand(Config config, HelpSystemHelper helper) {
+        super("help-thread", "Help thread specific commands", SlashCommandVisibility.GUILD);
+
+        OptionData category = new OptionData(OptionType.STRING, CATEGORY, "new category", true);
+        config.getHelpSystem()
+            .getCategories()
+            .forEach(categoryText -> category.addChoice(categoryText, categoryText));
+
+        getData().addSubcommandGroups(
+                new SubcommandGroupData("change", "Change the details of this help thread")
+                    .addSubcommands(
+                            new SubcommandData(CATEGORY, "Change the category of this help thread")
+                                .addOptions(category),
+                            new SubcommandData(TITLE, "Change the title of this help thread")
+                                .addOption(OptionType.STRING, TITLE, "new title", true)));
+
+        getData().addSubcommands(new SubcommandData(CLOSE, "Close this help thread"),
+                new SubcommandData(BOOKMARK, "Sends a dm linking thins help thread"));
+
+        this.helper = helper;
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandInteractionEvent event) {
+        ThreadChannel helpThread = event.getThreadChannel();
+        if (helpThread.isArchived()) {
+            event.reply("This thread is already closed.").setEphemeral(true).queue();
+            return;
+        }
+
+        boolean isOnCooldown = false;
```

Comment:
Move the whole cooldown logic up here, or into an extra method. It shouldnt be mixed with the stuff below.